src/sentry/static/sentry/app/components/asyncComponent.tsx

```diff
@@ -13,7 +14,25 @@ import PermissionDenied from 'app/views/permissionDenied';
 import RouteError from 'app/views/routeError';
 import getRouteStringFromRoutes from 'app/utils/getRouteStringFromRoutes';
 
-export default class AsyncComponent extends React.Component {
+type Props = {
+  location?: Location;
+  router?: any;
+  params?: any;
+};
+
+export type AsyncComponentState = {
+  loading: boolean;
+  reloading: boolean;
+  error: boolean;
+  errors: object;
+  remainingRequests?: number;
+  [key: string]: any;
+};
+
+export default class AsyncComponent<
+  P extends Props = Props,
+  S extends AsyncComponentState = AsyncComponentState
+> extends React.Component<P, S> {
   static propTypes = {
```

Comment:
Leaving the prop types in here until the usage of this component is fully converted to TS